Userland/Libraries/LibCore/ArgsParser.h

```diff
@@ -119,6 +126,9 @@ class ArgsParser {
     void add_positional_argument(double& value, char const* help_string, char const* name, Required required = Required::Yes);
 
 private:
+    template<bool RequireArgument = true>
+    void add_option_helper(char const* help_string, char const* long_name, char short_name, char const* value_name, OptionHideMode hide_mode, Function<bool(char const*)> accept_value);
```

Comment:
Again, please (1) use normal variable naming for non-type template arguments, and (2) create a bool-based enum to make all the calls to this function more readable.

Reply:
And the helper can easily receive Strings, no need to keep around old cruft if you're refactoring anyways :^). Now that I see that, please try to do that everywhere else as well, as far as it's possible with the many users.